research-software-directory/RSD-as-a-service
# feat: add testimonials to project pages

# Add testimonials to project pages

Closes #891

Changes proposed in this pull request:
* Added testimonial section to project pages
* Refactored software testimonials components and api to be able to share it with project      
* Added project testimonials to data generation

How to test:
* `make start` to build and create test data
* login as rsd admin in order to be able to edit existing data
* select software item for edit
* navigate to testimonials section. 
* add, update and remove testimonials. Confirm this works. Try to remove all items too.
* select project to edit and navigate to testimonials section.
* add, update and remove testimonials. Confirm this works. Try to remove all items too.
* create new software and add some testimonials to it.
* create new project and add some testimonials to it.

## Edit software testimonials
![image](https://github.com/user-attachments/assets/a8dc3573-a0cc-4a0e-b4d8-f742b46328c0)

## Software testimonials
![image](https://github.com/user-attachments/assets/e34fbbb0-805a-4381-a507-ad56dcaa9ab0)

## Edit project testimonials
![image](https://github.com/user-attachments/assets/4048f7a4-e5db-4bd0-87a4-218666c2f0cf)

## Project testimonials
![image](https://github.com/user-attachments/assets/b198e25e-23b4-4a71-8514-bc1cf408faf6)

## Delete testimonial modal
![image](https://github.com/user-attachments/assets/1bcdc55a-1728-4ffb-946c-1f5015bfb3f3)


PR Checklist:
